### ISSUE TITLE
Time formatting bug in DM3 files

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1299]
Some of my DM3 files appear to have the time formatting as `%d-%m-%Y` rather than `%m/%d/%Y`. The %m is a string, too. Not sure how to best handle this.

```
/Users/thomas/Dropbox/0_Git/GitHub/hyperspy_aar/hyperspy/io_plugins/digital_micrograph.py in _get_date(self, date)
    790 
    791     def _get_date(self, date):
--> 792         dt = datetime.strptime(date, "%m/%d/%Y")
    793         return dt.date().isoformat()
    794 

/Users/thomas/anaconda/envs/sims/lib/python3.5/_strptime.py in _strptime_datetime(cls, data_string, format)
    508     """Return a class cls instance based on the input string and the
    509     format string."""
--> 510     tt, fraction = _strptime(data_string, format)
    511     tzname, gmtoff = tt[-2:]
    512     args = tt[:6] + (fraction,)

/Users/thomas/anaconda/envs/sims/lib/python3.5/_strptime.py in _strptime(data_string, format)
    341     if not found:
    342         raise ValueError("time data %r does not match format %r" %
--> 343                          (data_string, format))
    344     if len(data_string) != found.end():
    345         raise ValueError("unconverted data remains: %s" %

ValueError: time data '17-May-16' does not match format '%m/%d/%Y'
```

**Comment by thomasaarholt** [comment]
Annoyingly I'll need to find out whether May should take the %B or %b formatter in `datetime`. %B is the full name and %b is the three-letter version, which for May is the same...

**Comment by thomasaarholt** [comment]
Actually, this is pretty cool:

``` python
>>> datestring = ("17/May/16")
>>> import dateutil.parser
>>> dateutil.parser.parse(datestring)
datetime.datetime(2016, 5, 17, 0, 0)
>>> print(...)
2016-05-17 00:00:00
```

**Comment by ericpre** [comment]
Fix in #1300 and #1282.

**Comment by francisco-dlp** [comment]
#1312
